ledger-core/virtual/statemachine/dispatcher.go

```diff
@@ -120,6 +120,19 @@ type errUnknownPayload struct {
 }
 
 func (c *conveyorDispatcher) Process(msg *message.Message) error {
+	ctx, logger := inslogger.WithTraceField(context.Background(), msg.Metadata.Get(defaults.TraceID))
+	defer func() {
+		if rec := recover(); rec != nil {
+			logger.Error(throw.R(rec, throw.E("ConveyorDispatcher.Process panic")))
+		}
+	}()
+	if err := c.process(ctx, msg); err != nil {
+		logger.Error(err)
+	}
+	return nil
```

Comment:
@MikhailZhukov Why we do not return an error in case if it occurred?

Reply:
watermill will infinitely retry message which raise error. 
`github.com/!three!dots!labs/watermill@v1.0.2/pubsub/gochannel/pubsub.go#L360`
```
		case <-msgToSend.Nacked():
			s.logger.Trace("Nack received, resending message", logFields)
			continue SendToSubscriber
```